pkg/controller/miganalytic/miganalytics_controller.go

```diff
@@ -275,6 +287,75 @@ func (r *ReconcileMigAnalytic) analyze(analytic *migapi.MigAnalytic) error {
 	return nil
 }
 
+func (r *ReconcileMigAnalytic) analyzeExtendedPVCapacity(client compat.Client, ns *migapi.MigAnalyticNamespace) error {
+	return nil
+}
+
+func (r *ReconcileMigAnalytic) getNodeToPVMapForNS(ns *migapi.MigAnalyticNamespace) (map[string][]persistentVolumeDetails, error) {
+
+	podList := corev1.PodList{}
+	nodeToPVDetails := map[string][]persistentVolumeDetails{}
+
+	err := r.Client.List(
+		context.TODO(),
+		k8sclient.InNamespace(ns.Namespace),
+		&podList)
+
+	if err != nil {
+		return nil, liberr.Wrap(err)
+	}
+
+	if len(podList.Items) > 0 {
+		for _, pod := range podList.Items {
+			if pod.Status.Phase == migapi.Running {
+				for _, vol := range pod.Spec.Volumes {
+					if vol.PersistentVolumeClaim != nil {
+						pvcObject := kapi.PersistentVolumeClaim{}
+						err := r.Client.Get(
+							context.TODO(),
+							types.NamespacedName{
+								Name:      vol.PersistentVolumeClaim.ClaimName,
+								Namespace: pod.Namespace,
+							}, &pvcObject)
+
+						if err != nil {
+							return nil, liberr.Wrap(err)
+						}
+
+						if volDetailList, exists := nodeToPVDetails[pod.Spec.NodeName]; exists {
+							volDetailList = append(volDetailList,
+								persistentVolumeDetails{
+									Name:                vol.PersistentVolumeClaim.ClaimName,
+									Namespace:           pvcObject.Namespace,
+									RequestedCapacity:   pvcObject.Spec.Resources.Requests.StorageEphemeral(),
+									PodUID:              pod.UID,
+									ProvisionedCapacity: pvcObject.Status.Capacity.StorageEphemeral(),
+									StorageClass:        pvcObject.Spec.StorageClassName,
+									VolumeName:          pvcObject.Spec.VolumeName,
+								})
+							nodeToPVDetails[pod.Spec.NodeName] = volDetailList
+						} else {
+							nodeToPVDetails[pod.Spec.NodeName] = []persistentVolumeDetails{{
+								Name:                vol.PersistentVolumeClaim.ClaimName,
+								Namespace:           pvcObject.Namespace,
+								RequestedCapacity:   pvcObject.Spec.Resources.Requests.StorageEphemeral(),
+								PodUID:              pod.UID,
+								ProvisionedCapacity: pvcObject.Status.Capacity.StorageEphemeral(),
+								StorageClass:        pvcObject.Spec.StorageClassName,
+								VolumeName:          pvcObject.Spec.VolumeName,
+							}}
+						}
+
+					}
+				}
+			}
+		}
+		return nodeToPVDetails, nil
+	}
+
+	return nil, nil
```

Comment:
Same as above, better to return a non-nil empty map

Reply:
Done